x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/streams/wired_stream.ts

```diff
@@ -0,0 +1,610 @@
+      //     index: this._definition.name,
+      //   });
+
+      //   return {
+      //     isValid: false,
+      //     errors: [
+      //       `Cannot create wired stream "${this._definition.name}" due to conflict caused by existing index`,
+      //     ],
+      //   };
+      // } catch (error) {
+      //   if (!isNotFoundError(error)) {
+      //     throw error;
+      //   }
+      // }
+    }
+
+    // validate routing
+    const children: Set<string> = new Set();
+    for (const routing of this._definition.ingest.wired.routing) {
+      if (children.has(routing.destination)) {
+        return {
+          isValid: false,
+          errors: [
+            `Child ${routing.destination} exists multiple times as child of ${this._definition.name}`,
+          ],
+        };
+      }
+      if (!isChildOf(this._definition.name, routing.destination)) {
+        return {
+          isValid: false,
+          errors: [
+            `The ID of child stream ${routing.destination} must start with the parent's name (${this._definition.name}), followed by a dot and a name`,
+          ],
+        };
+      }
+      children.add(routing.destination);
+    }
+
+    for (const stream of desiredState.all()) {
+      if (
+        !stream.isDeleted() &&
+        isChildOf(this._definition.name, stream.definition.name) &&
+        !children.has(stream.definition.name)
+      ) {
+        return {
+          isValid: false,
+          errors: [
+            `Child stream ${stream.definition.name} is not routed to from its parent ${this._definition.name}`,
+          ],
+        };
+      }
+    }
+
+    await this.assertNoHierarchicalConflicts(this._definition.name);
+
+    const [ancestors, descendants] = await Promise.all([
+      this.dependencies.streamsClient.getAncestors(this._definition.name),
+      this.dependencies.streamsClient.getDescendants(this._definition.name),
+    ]);
+
+    validateAncestorFields({
+      ancestors,
+      fields: this._definition.ingest.wired.fields,
+    });
+
+    validateDescendantFields({
+      descendants,
+      fields: this._definition.ingest.wired.fields,
+    });
+
+    const startingStateRoot = startingState.get(this._definition.name);
+    if (isRootStreamDefinition(this._definition) && startingStateRoot) {
+      // only allow selective updates to a root stream
+      validateRootStreamChanges(
+        startingStateRoot.definition as WiredStreamDefinition,
+        this._definition
+      );
+    }
+
+    validateSystemFields(this._definition);
+
+    if (this.dependencies.isServerless && isIlmLifecycle(this.getLifeCycle())) {
+      return { isValid: false, errors: ['Using ILM is not supported in Serverless'] };
+    }
+
+    return { isValid: true, errors: [] };
+  }
+
+  protected async doValidateDeletion(
+    desiredState: State,
+    startingState: State
+  ): Promise<ValidationResult> {
+    return { isValid: true, errors: [] };
```

Comment:
Just a note: This makes me think adding the "force=false" query param is not the only change we need to do - here the validation hook relies on the cascading changes doing its thing, which is probably OK for now, but we will need to check through all the validation logic whether it relies on it this way.